src/goto-analyzer/get_function_pointer_values.cpp

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************\
+
+Module: Show function pointer values
+
+Author: Diffblue Ltd, 2020
+
+\*******************************************************************/
+
+#include "get_function_pointer_values.h"
+
+#include <util/message.h>
+#include <util/namespace.h>
+
+#include <goto-programs/goto_model.h>
+#include <goto-programs/restrict_function_pointers.h>
+
+#include <analyses/ai.h>
+#include <analyses/variable-sensitivity/abstract_enviroment.h>
+#include <analyses/variable-sensitivity/value_set_abstract_object.h>
+#include <analyses/variable-sensitivity/variable_sensitivity_domain.h>
+
+#include <regex>
+#include <string>
+
+void get_function_pointer_values(
+  const goto_modelt &goto_model,
+  const ai_baset &ai,
+  const optionst &options,
+  message_handlert &message_handler,
+  std::string filename)
+{
+  function_pointer_restrictionst::restrictionst map;
+  namespacet ns(goto_model.symbol_table);
+
+  messaget m(message_handler);
+  m.status() << "Getting function pointer targets at specific callsites"
+             << messaget::eom;
+
+  for(auto const &function_entry : goto_model.goto_functions.function_map)
```

Comment:
This kind of data processing is where make_range comes into its own, and I'd strongly encourage taking some time to convert this to be a sequence of transformations